src/v/kafka/server/request_context.h

```diff
@@ -323,6 +324,11 @@ class request_context {
         auto result = do_authorized(operation, name, quiet);
         auto resp = bool(result);
 
+        // Not auditing authz attempts from audit principal
+        if (result.principal == security::audit_principal) {
+            return resp;
+        }
```

Comment:
Is it worth restricting this check? Are there operations that the audit principal needs to perform other than write the auth topic?

Reply:
Fair question, maybe this is a bit too heavy handed of an approach.  I think the more appropriate action may be to not audit produce actions for this user.  That would prevent auditing on add partitions to txn, end txn, init producer id, and produce for this user.